### PR TITLE
UAR-1288 Show correct confirmation details for a Remove journey

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1288

### Change description

* Delivered and working for the 'no change' journey
* Controller code changes should ensure only template changes are needed for the 'change' journey tickets
* Tests added and updated where appropriate

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/confirmation.controller.ts [hunk 1]
```diff
@@ -17,8 +19,24 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
     const appData: ApplicationData = getApplicationData(req.session);
     const referenceNumber = appData[Transactionkey];
 
+    // It's necessary to do this check (and save the result) before deleting the application data
```
Comment (markpit) [inline src/controllers/update/confirmation.controller.ts:22]: update the comment to say why - say app data used by isRemoveJourney

File: views/confirmation.html [hunk 2]
```diff
@@ -61,9 +67,15 @@ <h3 class="govuk-heading-s">What happens if the application is rejected</h3>
...
 
-      <p class="govuk-body">
-        <a class="govuk-link" href={{ "https://www.smartsurvey.co.uk/s/oe-update-conf/" if isUpdate else "https://www.smartsurvey.co.uk/s/roe-confirmation/" }}>What did you think of this service?</a> (takes 30 seconds)
-      </p>
+      {% if isUpdate %}
+        <p class="govuk-body">
+          <a class="govuk-link" href={{ "https://www.smartsurvey.co.uk/s/oe-remove-confirmation/" if (journey == "remove") else "https://www.smartsurvey.co.uk/s/oe-update-conf/" }}>What did you think of this service?</a> (takes 30 seconds)
+        </p>
+      {% else %}
+        <p class="govuk-body">
```
Comment (markpit) [inline views/confirmation.html:75]: looks like the else part of this is new but will also be shown if registration journey , is that correct?

Comment (mwestacott) [reply views/confirmation.html:75]: No, tested all scenarios (reg, update and remove) and screen shows correctly.

File: src/controllers/update/confirmation.controller.ts (hunk 1 above)
Comment (mwestacott) [reply src/controllers/update/confirmation.controller.ts]: Done.

File: views/confirmation.html [hunk 3]
```diff
@@ -41,7 +41,13 @@ <h2 class="govuk-heading-m">What you need to do now</h2>
       <h2 class="govuk-heading-m">What happens next</h2>
 
       {% if isUpdate %}
-        {% if isAgentRegistering or noChange %}
+        {% if (journey == "remove") %}
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply views/confirmation.html:44]: As discussed, going to refactor this (controller and template code) to improve readability.

**Review by markpit — APPROVED** [review]

**Pull request merged**